src/lib/journey/callbacks/select-idp/select-idp.mock.ts

```diff
@@ -1,3 +1,39 @@
+export default {
+  authId:
+    'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhdXRoSW5kZXhWYWx1ZSI6IlNvY2lhbExvZ2luLXdlYiIsIm90ayI6InU1dm0zODJzdDMxc3A1cWRmMGtvNm5lcmMxIiwiYXV0aEluZGV4VHlwZSI6InNlcnZpY2UiLCJyZWFsbSI6Ii9hbHBoYSIsInNlc3Npb25JZCI6IipBQUpUU1FBQ01ESUFCSFI1Y0dVQUNFcFhWRjlCVlZSSUFBSlRNUUFDTURFLipleUowZVhBaU9pSktWMVFpTENKamRIa2lPaUpLVjFRaUxDSmhiR2NpT2lKSVV6STFOaUo5LlpYbEtNR1ZZUVdsUGFVcExWakZSYVV4RFNteGliVTFwVDJsS1FrMVVTVFJSTUVwRVRGVm9WRTFxVlRKSmFYZHBXVmQ0YmtscWIybGFSMng1U1c0d0xpNUpjRlpRTkVzd2FETm9Oblp0Vm5GeVNYbFNRMFpCTGtOQ1ZqZDZXRlZYZUVScVZGOVdObXRaWkhjd1lucDRWalI1VVV0emRGTnNjekZ0ZERGU1NWZHVlVTlZWmtWTmJEWTRhMUV4VHpWb1RERmtObTVPT0RrMFdWZDZUamd3ZVVOQlFWQXdURmh4VFRGTFdGaElPWGhVUm1wMVZtVjZZVEZUVWtocGRFd3dka3h0YWxWTmNtVldTMWhHVTJoWlVDMXZTRGx3WTFsTk5qZHhNa0o1VW5KbGQwODBOVFZ0TUhob1JVeFBjMk5QTVZKNFFUZERXSEIyWjJSSVpFaG1PRTlJUjBwcGN6ZHFRWGhRUlhFMFF6ZHhlV1I2UW1GUVNqWkdlakJsWTNoQlduYzFXR3BmTUU1T1puQjRhbEJQVjNCVFZrOWZRVmsyVUdzeGJtRjBOV3hNVVdwWU1sQTVSVTFzV0RGUmFqUmtUams0WTBsa1VsOWxUa3hwUXpOdVpFaFNSMjlPY3pKVWFGSm5lRGgwUWxoNWVWRldVMVJOVDBSNmNrMHRZVVJUVW5CUU1HNWpObkkxTXpWV1ozUTVaRVJqVnpFMFUyZHBibTF3WWxSUVpYa3pUMkp4VlhKRE5tMDNlVEJvYmw4MmJFRmlWMkl5YUdSRVFreFFORW8wTmxGYVVqUjFRVFE1U1hadFQxWlBhMkZ0ZUZWcmJHRjZSbEZKU2tWU2N6ZFNUblV5VVVWeE5HMXNNM0pFWXkxYVZrVlZSRVZYVUhCMU4wODBRek5WVmtwR09GUjBNVU5uWlhnemNHSllYMmRyT0ZwNWJYZFRURzB0YTJGTFYwTkVOMXBNVTJaVFdrcERNSGxtVUZOeE1XdFBObGhLT1dGMVNGZHVRWHBSVGs5WGJWZFBPRkp4VDAxTGJXNDBSblZRUkhsdmVqbGZZMlZIU1cwNU5XUkpRa055ZEZScWVsUmhhelpsY0dsZlNYQlJYemN0TmxNd2FtNXBVazlUYUZGamNYVmhiVEUxV1ZGRWQwTTJVbkZoVldaSVVYTmFZbTVyVDNwSFNGaEhjM2RuU0V3eWVFUnhWa3BKUWpGT1RrRndTM1JEY25wc1VXczBiM2xtZG01d05ucFpTV3h2YkRGU2ExbzFVbFJLY0RCTmIzSlhZWGxzZVZBdE1EWXlaR3d6YW5obWVWZE5hbkJLY0dKVlZ6SlBiazlvUW0xRWVreEpXSE41T0hsUUxWRkdjVGw2YUhOZlNrOWhSa0p3UVVsNlIxQmFaSHBCT0MxM0xrWkVTVkUxV0VWVk9XaE9PSGMwUTI1a1FWbFRWWGMuX2RSUk40bGhDVzh1TlBpU2s4UU1LanVVSDJmRktkNHZHcnhKWE1sUFBYSSIsImV4cCI6MTY2OTA0OTEwNSwiaWF0IjoxNjY5MDQ4ODA1fQ.E8aTkN5BwRxhwQd_GnjOQ8V1V_7KvL50uIW7BWPadXM',
+  callbacks: [
+    {
+      type: 'SelectIdPCallback',
+      output: [
+        {
+          name: 'providers',
+          value: [
+            {
+              provider: 'apple-web',
+              uiConfig: {
+                buttonCustomStyle:
+                  'background-color: #000000; color: #ffffff; border-color: #000000;',
+                buttonImage: 'images/apple-logo.png',
+                buttonClass: '',
+                buttonCustomStyleHover:
+                  'background-color: #000000; color: #ffffff; border-color: #000000;',
+                buttonDisplayName: 'Apple',
+                iconClass: 'fa-apple',
+                iconFontColor: 'white',
+                iconBackground: '#000000',
+              },
+            },
+            { provider: 'localAuthentication' },
```

Comment:
Let's ensure we test the 3 different "modes" for social login:

1. Social IdPs only (buttons for Social IdPs only)
2. Social IdPs and local auth, but no other callbacks present (buttons for social and button for local, no form)
3. Social IdPs and local auth with username-password callbacks present (buttons for social only, button for local is not present, and form for immediately entering username and password)

We should have these different states tested as different stages in Storybook as well as our end-to-end tests. Let me know if you need any additional clarification.

Reply:
I found late last week that mode "2" in this list is a bit of a grey area, in that the expected behaviour based on platform UI and implied by the docs, is that if local auth is enabled but no other callbacks are present, then there is no button for local and the local auth option is ignored. I have therefore implemented the widget, storybook and e2e this way and intend to raise a docs JIRA to make it more obvious that for local auth to work, there need to be other callbacks in the page node. @ryanbas21 @cerebrl tagging you for visibility in case you want to challenge this.

Reply:
I'd like to challenge this, but let's not let it block the PR. When I originally worked on Social Login, the team (Bela at the time included) agreed to have a button for local authentication, which just allowed the journey designer to route that selection to a login form of the choosing. Again, we don't have to do it now, but I think this is needed for 1.0.

Reply:
Ryan and agreed that for this story just following what Platform UI does is sufficient. I am happy for it to be included in a future story as it took me by surprise to find that platform UI doesn't render it - it seems illogical to have an outcome on the node that you can't get to by building your tree a certain way. Having 'local auth' route to a login form makes sense to me. I'll put it in another story. Thanks @cerebrl.

Reply:
Created [SDKS-2250](https://bugster.forgerock.org/jira/browse/SDKS-2250) for this.